protocol/messenger_backup_handler.go

```diff
@@ -113,6 +114,10 @@ func (m *Messenger) handleBackedUpProfile(message *protobuf.BackedUpProfile, bac
 		Profile: &wakusync.BackedUpProfile{},
 	}
 
+	if err := utils.ValidateDisplayName(&message.DisplayName); err != nil {
+		return err
```

Comment:
I am wondering whether this should really be an error. Users with invalid display name won't be able to recover the account. I would go with:

```golang
if err == nil {
    err := m.SaveSyncDisplayName(message.DisplayName, message.DisplayNameClock)
    ...
} else {
   m.logger.Warn(...)
}
```

Reply:
I don't think it was addressed :thinking:

Reply:
Ah lol, I did the change in `messenger.go` as I thought you were saying that the problem was with the login :facepalm:

Reply:
@osmaczko alright I updated for real this time. I even added a new test to validate the behaviour.

Let me know what you think of the change in `messenger.go` too. Should we fail the login if it fails?

Reply:
https://github.com/status-im/status-go/pull/5392#discussion_r1665411273
https://github.com/status-im/status-go/pull/5392#discussion_r1665412554